### PR TITLE
Pbi 138 139: integrate SES as Email Service Provider

### PR DESCRIPTION
**Pull request by valeelim**
## Describe your changes
- Implement abstraction for SES
- Unit test for the SES implementation
- Setup AWS config dependency

## Issue ticket number and link
- [ADUDU-138](https://linear.app/adudu/issue/ADU-138/api-integration-with-ses-provider)
- [ADUDU-139](https://linear.app/adudu/issue/ADU-139/[unit-test]-for-ses)

### CODE REVIEW

**Review by edutjie — APPROVED** [review]
LGTM

File: cmd/config/config.go [hunk 1]
```diff
@@ -21,6 +22,12 @@ type SMTP struct {
 	AuthPassword string `mapstructure:"auth_password" validate:"required"`
 }
 
+type AWS struct {
+	AccessKey       string `mapstructure:"access-key" validate:"required"`
+	SecretAccessKey string `mapstructure:"secret-access-key" validate:"required"`
+	Region          string `mapstructure:"region" validate:"required"`
+}
+
```
Comment (ferryhieronimus) [inline cmd/config/config.go:30]: What if we use multiple AWS services with different region?

Comment (valeelim) [reply cmd/config/config.go:30]: we wont 😂

Comment (valeelim) [reply cmd/config/config.go:30]: even so, ideally the access should be granted by CLI through bash scripts, not on the App layer
hence the solution, but this is just a simple one cause setting up the script might introduce some unnecessary complexity for our repo

**Pull request merged**